### DISCUSSION TITLE
GithubApp 403 resource not accessible by integration with Githubapp

### DISCUSSION THREAD

**Discussion by ghost** [community/community#45524] [Actions]
Hi. I am developing following actions to github. I have created Github actions workflows for GitHub repositories creation, archieving and deletion. I have managed to get PyGithub python scripts work with finegrained tokens, but some reason GitHubapp tokens are not working even repo administator write/read permission are granted. Still getting 403 resource not accessible by integration. 

Have any one have idea what could be the problem?

**Comment by jge162** [comment]
reason could be that the token you are using is not valid or has expired.

**Reply by jge162** [reply]
[document](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)

Here is documentation on token. I believe this is where your issue is derived.

**Reply by jge162** [reply]
It’s also discussed here #[8843 ](https://github.com/github/codeql/issues/8843)

**Reply by ghost** [reply]
Ok. Thanks I will recheck this

**Reply by jge162** [reply]
If you resolve issue please let us know. It could help someone else down the road. Thank you.

**Reply by ghost** [reply]
More details for this I am using tibdex/github-app-token to generate token. And i have granted permissions to administration - read write and metadata - read. Reviewed app permissions and installed in use. Using pygithub api for creating repositories, but still get 403.

**Reply by jge162** [reply]
403 error usually indicates that you are unauthorized to perform the requested action. Make sure that you have the correct permissions to create repositories with the Github API. ->`check if you have provided the correct Github App token, and if it has the necessary permissions (administration - read, write and metadata - read) to perform the action.`

**Reply by jge162** [reply]
try regenerating the token

**Reply by ghost** [reply]
Ok. Thanks for fast reply. Just tested setup new client secret, update action secrets but no luck. Tibdex action generates token based on private key and appid. Token generation passes but repo creation do not. Also set all permissions on but still not working. I am using manual workflow_dispatch to execute pipeline.

**Reply by ghost** [reply]
Ok. Found the reason. I used getUser().createrepo and should use org.createrepo() at ci side.

**Reply by jge162** [reply]
Nice
